src/compose/ComposeBox.android.js

```diff
@@ -0,0 +1,378 @@
+/* @flow */
+import React, { PureComponent } from 'react';
+import { View, TextInput, findNodeHandle } from 'react-native';
+import { connect } from 'react-redux';
+import TextInputReset from 'react-native-text-input-reset';
+import isEqual from 'lodash.isequal';
+
+import type {
+  Auth,
+  Context,
+  Narrow,
+  EditMessage,
+  InputSelectionType,
+  User,
+  Dispatch,
+  Dimensions,
+  GlobalState,
+} from '../types';
+import {
+  addToOutbox,
+  cancelEditMessage,
+  draftAdd,
+  draftRemove,
+  fetchTopicsForActiveStream,
+  sendTypingEvent,
+} from '../actions';
+import { updateMessage } from '../api';
+import { FloatingActionButton, Input, MultilineInput } from '../common';
+import { showErrorAlert } from '../utils/info';
+import { IconDone, IconSend } from '../common/Icons';
+import { isStreamNarrow, isStreamOrTopicNarrow, topicNarrow } from '../utils/narrow';
+import ComposeMenu from './ComposeMenu';
+import AutocompleteViewWrapper from '../autocomplete/AutocompleteViewWrapper';
+import getComposeInputPlaceholder from './getComposeInputPlaceholder';
+import NotSubscribed from '../message/NotSubscribed';
+
+import {
+  getAuth,
+  getSession,
+  canSendToActiveNarrow,
+  getLastMessageTopic,
+  getActiveUsersAndBots,
+  getShowMessagePlaceholders,
+} from '../selectors';
+import { getIsActiveStreamSubscribed } from '../subscriptions/subscriptionSelectors';
+import { getDraftForActiveNarrow } from '../drafts/draftsSelectors';
+
+type Props = {
+  auth: Auth,
+  canSend: boolean,
+  narrow: Narrow,
+  usersAndBots: User[],
+  draft: string,
+  lastMessageTopic: string,
+  isSubscribed: boolean,
+  editMessage: EditMessage,
+  safeAreaInsets: Dimensions,
+  dispatch: Dispatch,
+  messageInputRef: (component: any) => void,
+};
+
+type State = {
+  isMessageFocused: boolean,
+  isTopicFocused: boolean,
+  isMenuExpanded: boolean,
+  topic: string,
+  message: string,
+  height: number,
+  selection: InputSelectionType,
+};
+
+export const updateTextInput = (textInput: TextInput, text: string): void => {
+  if (!textInput) {
+    return;
+  }
+
+  textInput.setNativeProps({ text });
```

Comment:
This line appears in the commit that first introduces `updateTextInput`, while the inputs are still controlled. I'm still a bit confused about exactly what it does.

In the case where the input is controlled, is it your understanding that this line has an effect on behavior? Or does it not have an effect for some reason?

If it doesn't have an effect, I'd appreciate an explanation of that in the commit message -- or even just a mention of it. Even just knowing that that was your intention when writing the commit is very helpful for trying to understand it.

If it does have an effect on behavior, what is that effect?